### DISCUSSION TITLE
0 features are filtered!

### DISCUSSION THREAD

**Discussion by niall142** [cafferychen777/ggpicrust2#119] [Q&A] [answered]
Hi all, I'm having my first go trying to run ggpicrust2 with some bacterial abundance data from an orchard. I'm inputting the abundance data with the species names to identify the rows. My metadata file is identifying the treatment group for each sample.

I'm trying to replicate the example from the CRAN README page. 

Here's the code I'm using:

```
results_file_input <- ggpicrust2(data = SEM.bacteria.cpg.2024.rename.test,
                                 metadata = SEM.soil.samples.IDs.2024 ,
                                 group = "mulch.treatment", 
                                 pathway = "KO",
                                 daa_method = "LinDA",
                                 reference = "con",
                                 ko_to_kegg = TRUE,
                                 order = "pathway_class",
                                 p_values_bar = TRUE,
                                 x_lab = "pathway_name")
```

And here's the console readout I get when I run this code:

```
Starting the ggpicrust2 analysis...

Converting KO to KEGG...

Processing provided data frame...
Loading KEGG reference data. This might take a while...
Performing KO to KEGG conversion. Please be patient, this might take a while...
  |================================================================================================================================================================================================================| 100%
KO to KEGG conversion completed. Time elapsed: 18.82 seconds.
Removing KEGG pathways with zero abundance across all samples...
KEGG abundance calculation completed successfully.
Performing pathway differential abundance analysis...

Sample names extracted.
Identifying matching columns in metadata...
Matching columns identified: plot.ID . This is important for ensuring data consistency.
Using all columns in abundance.
Converting abundance to a matrix...
Reordering metadata...
Converting metadata to a matrix and data frame...
Extracting group information...
Running LinDA analysis...
Performing LinDA analysis...
Registered S3 methods overwritten by 'rmutil':
  method         from 
  plot.residuals psych
  print.response httr 
0  features are filtered!
The filtered data has  0  samples and  0  features will be tested!
Fit linear models ...
Error in x[, ii] : subscript out of bounds
```

Anyone know what's going wrong here?

Thanks!
-Niall

**Comment by cafferychen777** [comment]
Dear @niall142,

Thank you for reaching out about the issue you're experiencing with ggpicrust2. I appreciate you providing the code and console output, but without access to your actual data files, it's challenging to pinpoint the exact cause of the problem.

To help me debug this issue more effectively, would you be willing to send your data files to cafferychen777@tamu.edu? Specifically, it would be helpful to have:

1. Your bacterial abundance data file (SEM.bacteria.cpg.2024.rename.test)
2. Your metadata file (SEM.soil.samples.IDs.2024)

With these files, I'll be able to reproduce the error and investigate what might be causing the "0 features are filtered" issue and the subsequent error.

Please ensure to remove any sensitive information from the files before sending them. Once I receive the data, I'll take a closer look and get back to you with more specific guidance.

Thank you for your patience and cooperation in resolving this issue.

Best regards,
Chen YANG

**Comment by niall142** [comment]
Hi Chen,

Thanks very much for your reply, I've attached my abundance data and my
metadata files here. As in the readme example, I've saved the abundance
data as a .tsv file, and the metadata as .txt, but I wasn't sure whether
that's a specific requirement of ggpicrust or not. I'm also not entirely
sure I have the right format; does it matter for picrust whether you use
copies-per-gram abundance data, or another form? Are the species names of
the bacteria sufficient for the database search?

Thanks!
-Niall

On Mon, Sep 23, 2024 at 12:15 AM Caffery Yang ***@***.***>
wrote:

> Dear @niall142 <https://github.com/niall142>,
>
> Thank you for reaching out about the issue you're experiencing with
> ggpicrust2. I appreciate you providing the code and console output, but
> without access to your actual data files, it's challenging to pinpoint the
> exact cause of the problem.
>
> To help me debug this issue more effectively, would you be willing to send
> your data files to ***@***.***? Specifically, it would be
> helpful to have:
>
>    1. Your bacterial abundance data file
>    (SEM.bacteria.cpg.2024.rename.test)
>    2. Your metadata file (SEM.soil.samples.IDs.2024)
>
> With these files, I'll be able to reproduce the error and investigate what
> might be causing the "0 features are filtered" issue and the subsequent
> error.
>
> Please ensure to remove any sensitive information from the files before
> sending them. Once I receive the data, I'll take a closer look and get back
> to you with more specific guidance.
>
> Thank you for your patience and cooperation in resolving this issue.
>
> Best regards,
> Chen YANG
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cafferychen777/ggpicrust2/discussions/119#discussioncomment-10722618>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARY3AF5M3YYOC5KOTUUS33LZX65XXAVCNFSM6AAAAABOSZVR52VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZSGI3DCOA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>

plot.ID	trial	row	mulch.treatment	mulch.treatment.flag	year	code	plot.ID.year
SEM.R1.con	SEM	R1	con	white	2024	CU600O	SEM.R1.con.2024
SEM.R1.new	SEM	R1	new	yellow	2024	CU600P	SEM.R1.new.2024
SEM.R1.old	SEM	R1	old	pink	2024	CU600Q	SEM.R1.old.2024
SEM.R2.old	SEM	R2	old	pink	2024	CU600R	SEM.R2.old.2024
SEM.R2.new	SEM	R2	new	yellow	2024	CU600S	SEM.R2.new.2024
SEM.R2.con	SEM	R2	con	white	2024	CU600T	SEM.R2.con.2024
SEM.R3.new	SEM	R3	new	yellow	2024	CU600U	SEM.R3.new.2024
SEM.R3.con	SEM	R3	con	white	2024	CU600V	SEM.R3.con.2024
SEM.R3.old	SEM	R3	old	pink	2024	CU600W	SEM.R3.old.2024
SEM.R4.new	SEM	R4	new	yellow	2024	CU600X	SEM.R4.new.2024
SEM.R4.old	SEM	R4	old	pink	2024	CU600Y	SEM.R4.old.2024
SEM.R4.con	SEM	R4	con	white	2024	CU600Z	SEM.R4.con.2024

**Reply by cafferychen777** [reply]
Dear Niall,

Thank you for your response and for attempting to share your data files. Unfortunately, it seems there was an issue with uploading the data to GitHub. The discussion thread only shows the metadata content, but the abundance data file is missing.

Given these limitations with GitHub, I suggest sending your data files directly to my email address: cafferychen777@tamu.edu. This will ensure I receive both the abundance data and the metadata files in their entirety.

Please include the following in your email:

1. The bacterial abundance data file (SEM.bacteria.cpg.2024.rename.test)
2. The metadata file (SEM.soil.samples.IDs.2024)

Once I receive these files, I'll be able to thoroughly examine your data and provide more specific guidance on resolving the "0 features are filtered" issue you're experiencing with ggpicrust2.

Thank you for your patience and cooperation. I look forward to helping you resolve this problem.

Best regards,
Chen Yang

**Comment by niall142** [comment]
Woops, my mistake, I sent the files to your email there now.

Thanks!
-Niall

**Reply by cafferychen777** [reply, marked as answer]
Dear Niall,

Thank you for sending your data files and for your patience as we work through this issue. After reviewing your files, I believe I've identified the source of the problem.

It appears there may have been a misunderstanding about the type of data ggpicrust2 is designed to work with. The package is specifically tailored for use with data generated by PICRUSt2, which provides functional predictions based on 16S rRNA gene sequences. The bacterial abundance data you've provided, while valuable, is not the type of input ggpicrust2 is optimized to process.

While it is possible to use ggpicrust2 with your data, it may not be the most efficient or appropriate tool for your specific analysis needs. Instead, I'd like to recommend an alternative that might be more suitable for your bacterial abundance data.

I suggest you take a look at the MicrobiomeStat package. This package offers both differential abundance analysis and visualization tools that are well-suited to the type of data you're working with. You can find more information about MicrobiomeStat at www.microbiomestat.wiki.

In particular, I think you might find the section on cross-sectional taxa analysis especially relevant to your needs. You can find this information here:
https://www.microbiomestat.wiki/single-point-analysis/cross-sectional-investigation-taxa-analysis-with-microbiomestat

This resource should provide you with guidance on how to perform the kind of analysis you're looking for, using tools that are more appropriate for your bacterial abundance data.

If you have any questions about using MicrobiomeStat or need further clarification, please don't hesitate to ask. I'm here to help you find the best solution for your research needs.

Best regards,
Chen Yang

**Comment by niall142** [comment]
Hi Chen,

Thanks very much for taking a look for me, that's a big help! I'll take a
look at MicrobiomeStat, thanks for the recommendation.

-Niall
